code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -162,6 +162,13 @@
 	else
 		affecting = get_bodypart(ran_zone(user.zone_selected))
 	var/target_area = parse_zone(check_zone(user.zone_selected)) //our intended target
+	if(I.force && I.damtype != STAMINA && affecting.status == BODYPART_ROBOTIC) // Bodpart_robotic sparks when hit, but only when it does real damage
+		var/datum/effect_system/spark_spread/spark_system = new /datum/effect_system/spark_spread
+		spark_system.set_up(1, 0, src)
+		spark_system.attach(src)
+		spark_system.start()
+		if(prob(25))
+			new /obj/effect/decal/cleanable/oil(src.loc)

```

Comment:
pretty sure `src.` is superfluous

Reply:
It is not. Removed and tested and it stops working.

Reply:
alright cool, I wasn't certain on this one. I assume it is getting confused and trying to take the implication from the new item.